airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py

```diff
@@ -92,42 +82,42 @@ def is_metrics_in_custom_query(query: str) -> bool:
     def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
         try:
             logger.info("Checking the config")
-            google_api = GoogleAds(credentials=self.get_credentials(config), customer_id=config["customer_id"])
-            account_info = self.get_account_info(google_api, config)
-            is_manager_account = self.is_manager_account(account_info)
+            google_api = GoogleAds(credentials=self.get_credentials(config))
 
+            accounts = self.get_account_info(google_api, config)
+            customers = Customer.from_accounts(accounts)
             # Check custom query request validity by sending metric request with non-existant time window
-            for query in config.get("custom_queries", []):
-                query = query.get("query")
-
-                if is_manager_account and self.is_metrics_in_custom_query(query):
-                    return False, f"Metrics are not available for manager account. Check fields in your custom query: {query}"
-                if CustomQuery.cursor_field in query:
-                    return False, f"Custom query should not contain {CustomQuery.cursor_field}"
-
-                req_q = CustomQuery.insert_segments_date_expr(query, "1980-01-01", "1980-01-01")
-                for customer_id in google_api.customer_ids:
-                    google_api.send_request(req_q, customer_id=customer_id)
+            for customer in customers:
+                for query in config.get("custom_queries", []):
+                    query = query.get("query")
+                    if customer.is_manager_account and self.is_metrics_in_custom_query(query):
+                        return False, f"Metrics are not available for manager account. Check fields in your custom query: {query}"
```

Comment:
we should put a note in the docs that mentions that custom queries must work for ALL account ids input. If i sync 9 non-manager accountss, and 1 manager account, the connection check will fail. Alternatively we can just skip that custom stream for the manager account and log a message to the user.

Reply:
> Alternatively we can just skip that custom stream for the manager account and log a message to the user.
Done

Reply:
Done https://github.com/airbytehq/airbyte/pull/12914/files#diff-04615dae4ed388b7d6ab2905249f18c69c801d1ab6de1d3d27bf2030491b7db5R115